pkg/kubectl/cmd/set/set_selector.go

```diff
@@ -0,0 +1,280 @@
+	if overwrite || dst.MatchExpressions == nil {
+		dst.MatchExpressions = make([]unversioned.LabelSelectorRequirement, len(src.MatchExpressions))
+	}
+	if overwrite {
+		copy(dst.MatchExpressions, src.MatchExpressions)
+		return nil
+	}
+	for _, srcExp := range src.MatchExpressions {
+		for _, dstExp := range dst.MatchExpressions {
+			if srcExp.Key == dstExp.Key {
+				return fmt.Errorf("MatchExpression %v exists, use --overwrite to replace it", srcExp.Key)
+			}
+		}
+	}
+	for _, srcExp := range src.MatchExpressions {
+		dst.MatchExpressions = append(dst.MatchExpressions, srcExp)
+	}
+	return nil
+}
+
+// Run the command.
+func (o *SelectorOptions) Run() error {
+	allErrs := []error{}
+
+	patches := CalculatePatches(o.Infos, o.Encoder, func(info *resource.Info) (bool, error) {
+		err := o.UpdateSelectorForObject(info.Object, func(map_selector *map[string]string, selector *unversioned.LabelSelector) error {
+			if map_selector != nil {
+				if len(o.Selector.MatchExpressions) > 0 {
+					return fmt.Errorf("error: MatchExpression %v not supported on this object", o.Selector.MatchExpressions)
+				}
+				err := copyMatchLabels(map_selector, &o.Selector.MatchLabels, o.Overwrite)
+				if err != nil {
+					return err
+				}
+			}
+			if selector != nil {
+				err := copySelector(selector, o.Selector, o.Overwrite)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		})
+		return true, err
+	})
+
+	for _, patch := range patches {
+		info := patch.Info
+		if patch.Err != nil {
+			allErrs = append(allErrs, fmt.Errorf("error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err))
+			continue
+		}
+		// no changes
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
+			continue
+		}
+
+		if o.Local || o.DryRun {
+			fmt.Fprintln(o.Out, "running in local/dryRun mode...")
+			return o.PrintObject(o.Cmd, o.Mapper, info.Object, o.Out)
+		}
+
+		// patch the change
+		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, api.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			allErrs = append(allErrs, fmt.Errorf("failed to patch selector update to template: %v\n", err))
+			continue
+		}
+		info.Refresh(obj, true)
+
+		// record this change (for rollout history)
+		if o.Record || cmdutil.ContainsChangeCause(info) {
+			if err := cmdutil.RecordChangeCause(obj, o.ChangeCause); err == nil {
+				if obj, err = resource.NewHelper(info.Client, info.Mapping).Replace(info.Namespace, info.Name, false, obj); err != nil {
+					allErrs = append(allErrs, fmt.Errorf("changes to %s/%s can't be recorded: %v\n", info.Mapping.Resource, info.Name, err))
+				}
+			}
+		}
+
+		info.Refresh(obj, true)
+		cmdutil.PrintSuccess(o.Mapper, o.ShortOutput, o.Out, info.Mapping.Resource, info.Name, "selector updated")
+	}
+	return utilerrors.NewAggregate(allErrs)
+}
+
+// getResourcesAndSelector retrieves resources and the selector expression from the given args (assuming selectors the last arg)
```

Comment:
I'm not super familiar with the new label selector, but I can see that the lexer has some logic around whitespace.  Do we have another example of entering it in the CLI?  I'm not entirely sure that "single arg at the end" is going to be easiest option given that we optionally specify resources here.